di2f/index.md

```diff
@@ -0,0 +1,226 @@
+
+By identifying a thread as a global resource, any peer can determine the following from its DID:
+
+1.  Who can write to the thread?
+2.  Who can read from the thread?
+3.  Where can the thread be bootstrapped from?
+4.  Who controls (1), (2), and (3)?
+
+It is important to note that a thread DID is an _identifier_, not an _identity_ in the usual sense. To illustrate, if a thread is a resource that a peer on the network is attempting to identify, i.e., the thread itself _is the subject_, and the thread document is a _representation of the subject_, then the DID _subject is not the controller_ [@reed2020decentralized]. The controller is one or more identity-based DIDs, e.g., `did:key:foo`, `did:ethr:foo`, `did:3:foo`, etc.
+
+A (proposed) thread DID document is structured as in [@lst:thread-did]. In this example, the controller (`did:key:foo`) is defined by a key-based DID [@longely2021didKey]. The controller is able to modify the thread DID document. Additionally, both `did:key:foo"` and `did:key:bar"` can authenticate as `did:thread:id`, meaning ... . Finally, the thread DID document in [@lst:thread-did] also specifies two `serviceEndpoint`s that outlines which peers (`/p2p/peer-id-1` and `/p2p/peer-id-2`) are able to bootstrap the thread.
+
+<!-- @sanderpick What does "authenticate as `"did:thread:id"`" mean in this context? Can we be more explicit here" -->
+
+<!-- @sanderpick What does "bootstrap" mean in this context? Can we be more explicit here" -->
+
+<div id="lst:thread-did" class="listing">
+Proposed thread DID Document structure.
+
+```{.json}
+{
+  "@context": "https://www.w3.org/ns/did/v1",
+  "id": "did:thread:id",
+  "controller": "did:key:foo",
+  "authentication": [
+    {
+      "id": "did:key:foo#keys-1",
+      "type": "Ed25519VerificationKey2018",
+      "controller": "did:key:foo",
+      "publicKeyBase58": "..."
+    },
+    {
+      "id": "did:key:bar#keys-1",
+      "type": "Ed25519VerificationKey2018",
+      "controller": "did:key:bar",
+      "publicKeyBase58": "..."
+    }
+  ],
+  "service": [
+    {
+      "id": "did:thread:<id>#peer-id-1",
+      "type": "threadService",
+      "serviceEndpoint": "/p2p/peer-id-1",
+      "serviceProtocol": "/thread/0.0.1"
+    },
+    {
+      "id": "did:thread:<id>#peer-id-2",
+      "type": "threadService",
+      "serviceEndpoint": "/p2p/peer-id-2",
+      "serviceProtocol": "/thread/0.0.1"
+    }
+  ]
+}
+```
+
+</div>
+
+# Method Definition
+
+The crux of any DID implementation is defining it's _DID method_. The DID specification defines a _method_ as a "means to implement this specification on different verifiable data registries" [@reed2020decentralized (sec 8)]. In other words, new DID methods should be specified such that different implementations of the same DID method remain interoperable.
+
+<!-- @sanderpick Maybe that last statement is "weak", drop it? Something better to put there? -->
```

Comment:
@sanderpick Maybe that last statement is "weak", drop it? Something better to put there?

Reply:
Maybe something like, "The core function of the spec is to ensure interoperability between different DID methods"?

Reply:
Love it!